pkg/apimachinery/utils/folder_key.go

```diff
@@ -0,0 +1,20 @@
+package utils
+
+import (
+	"context"
+)
+
+type folderRemovePermissions string
+
+const removePermissionKey = folderRemovePermissions("RemovePermission")
+
+func SetFolderRemovePermissions(ctx context.Context, value bool) context.Context {
+	return context.WithValue(ctx, removePermissionKey, value)
+}
+
+func GetFolderRemovePermissions(ctx context.Context, defaultVal bool) bool {
+	if val, ok := ctx.Value(removePermissionKey).(bool); ok {
+		return val
+	}
+	return defaultVal
+}
```

Comment:
NIT, this is usually the go–to way to use keys in context
```suggestion
type folderRemovePermissions struct{}

func SetFolderRemovePermissions(ctx context.Context, value bool) context.Context {
	return context.WithValue(ctx, folderRemovePermissions{}, value)
}

func GetFolderRemovePermissions(ctx context.Context, defaultVal bool) bool {
	if val, ok := ctx.Value(folderRemovePermissions{}).(bool); ok {
		return val
	}
	return defaultVal
}
```

Reply:
`string` or `int` are often used too if one needs to have multiple keys. Using `type folderRemovePermissions struct{}` allows single instance of that type only.

Reply:
> Using type `folderRemovePermissions struct{}` allows single instance of that type only.
True, but my understanding is that this is what we need?